crates/services/src/service.rs

```diff
@@ -361,14 +402,14 @@ async fn run_task<S: RunnableTask>(
         let result = tracked_result.extract(metric);
 
         match result {
-            Ok(should_continue) => {
-                if !should_continue {
-                    tracing::debug!("stopping");
-                    break;
-                }
+            TaskRunResult::Continue => {
                 tracing::debug!("run loop");
             }
-            Err(e) => {
+            TaskRunResult::Stop => {
+                tracing::debug!("stopping");
+                break;
+            }
+            TaskRunResult::ErrorContinue(e) => {
                 let e: &dyn std::error::Error = &*e;
                 tracing::error!(e);
```

Comment:
would it make sense to add a debug! statement to explicitly tell that the loop will continue (similar to the `Continue` case)?

Reply:
I'm not sure I understand. Are you talking in the `ErrorContinue` case? I feel like we should definitely use `error!` there, especially since we don't exit on errors.